kyaml/yaml/datamap.go

```diff
@@ -95,6 +95,35 @@ func makeSecretValueRNode(s string) *RNode {
 	return NewRNode(yN)
 }
 
+func (rn *RNode) LoadMapIntoSecretStringData(m map[string]string) error {
+	for _, k := range SortedMapKeys(m) {
```

Comment:
What is the significance of walking the keys in lexicographical order? Is that to avoid any nondeterminism? Since the key names must be unique, they can't conflict with one another in the "m" map, but perhaps we're wary of what the `SetField` function will do against the `RNode`'s existing fields. I'm still not sure if that's a problem, though.

If you walked the map by its keys and values together, you could avoid the index expression on the next line.